Alloy/Encoders/MaxEncoder.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  MaxEncoder.swift
+//  Adjuster
+//
+//  Created by Eugene Bokhan on 13/02/2019.
+//
+
+import Metal
+import simd
+
+public class MaxEncoder {
+
+    enum Errors: Error {
+        case functionCreationFailed
+        case encoderCreationFailed
+        case invalidBufferStorageMode
+    }
+
+    let pipelineState: MTLComputePipelineState
+
+    public init(library: MTLLibrary) throws {
+        guard
+            let function = library.makeFunction(name: MaxEncoder.functionName)
+        else { throw Errors.functionCreationFailed }
+        let pipelineState = try library.device.makeComputePipelineState(function: function)
+        self.pipelineState = pipelineState
+    }
+
+    public func encode(inputTexture: MTLTexture,
+                       blockSize: BlockSize,
+                       resultBuffer: MTLBuffer,
+                       in commandBuffer: MTLCommandBuffer) throws {
+        guard resultBuffer.storageMode == .shared
+        else { throw Errors.invalidBufferStorageMode }
```

Comment:
this is not correct

Reply:
you mean we need to add .managed for macOS cases?

Reply:
we don’t have to enforce any